kube-client/src/api/portforward.rs

```diff
@@ -293,6 +315,32 @@ where
                     .await
                     .map_err(Error::SendWebSocketMessage)?;
             }
+            Message::ToPodClose(ch) => {
+                let ch = ch as usize;
+                if ch >= initialized.len() {
+                    return Err(Error::InvalidChannel(ch));
+                }
+                let port_index = ch / 2;
+                if !shutdown[port_index] {
+                    writers[port_index].shutdown().await.map_err(Error::Shutdown)?;
+                    shutdown[port_index] = true;
+
+                    closed_ports += 1;
+                }
+            }
+            Message::FromPodClose => {
+                for writer in &mut writers {
+                    writer.shutdown().await.map_err(Error::Shutdown)?;
+                }
+            }
+        }
+
+        if closed_ports == ports.len() && !socket_shutdown {
```

Comment:
Do we need `socket_shutdown`? Can't we stop the loop?

Reply:
I'll have to check but IIRC that would mean we stop this worker task but not the `from_pod_loop` task, meaning the `join` would get stuck.

Reply:
Yep I've confirmed, if we just exit the loop we don't close the stream.